src/Components/App/App.js

```diff
@@ -43,8 +44,10 @@ const App = () => {
   }
 
   const favTruck = (truck) => {
-    const newFavState = [...favorites, truck]
-    setFavorites(newFavState)
+    if (!favorites.includes(truck)) {
```

Comment:
This took some time and brain power to work through these bugs, but you are just the person for the job! Thank you!